src/basic/copy.c

```diff
@@ -132,6 +130,10 @@ int copy_bytes(int fdf, int fdt, uint64_t max_bytes, bool try_reflink) {
                         assert(max_bytes >= (uint64_t) n);
                         max_bytes -= n;
                 }
+                /* sendfile accepts at most SSIZE_MAX-offset bytes to copy,
+                 * so reduce our maximum by the amount we already copied,
+                 * but always stay above 0. */
+                m = MAX(1u, m - n);
```

Comment:
I don't get it, please help me understand why this clamp is necessary. Moreover, what's the reason for not `break`ing here if `m == n`?